### PR TITLE
Add auto-release GitHub action.

### PR DESCRIPTION
**Pull request by ericmjl**
## PR Description

As per title. Should make releasing a new version of jax-unirep a bit easier.

### CODE REVIEW

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -1,60 +1,104 @@
...
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
 
+      - name: Pull latest commits of dev branch
+        run: |
+          git checkout dev
+          git pull
```
Comment (ElArkk) [inline .github/workflows/auto-release.yml:32]:
```suggestion
      - name: Pull latest commits of master branch
        run: |
          git checkout master
          git pull
```

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -1,60 +1,104 @@
...
+
+      # See https://github.com/marketplace/actions/wait-on-check
+      # TODO: Fix this later (have every push to dev trigger a test suite run)
+      # - name: Ensure all tests are passing on dev
+      #   uses: lewagon/wait-on-check-action@v0.2
+      #   with:
+      #     ref: dev  # can be commit SHA or tag
+      #     check-name: pyjanitor tests
```
Comment (ElArkk) [inline .github/workflows/auto-release.yml:40]: wrong repo 😆

**Review by ElArkk — COMMENTED** [review]
Cool stuff, thanks for taking this on!

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (ericmjl) [reply .github/workflows/auto-release.yml]: Bahhhh, haha, okay, now you know where the thing came from lol.

**Comment by ericmjl** [comment]
Yooooo! Ima gonna merge. This one shouldn't affect the software too much.

**Pull request merged**